sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1198,8 +1209,34 @@ def process(self, element, *schema_side_inputs):
       return self._flush_all_batches()
 
   def finish_bundle(self):
+    current_millis = int(time.time() * 1000)
```

Comment:
Are we sure finishBundle is called frequently enough? I.e. are there bundles which process for a long time? Are we sure this code will get invoked when there are failures to write to BQ?

Reply:
> Are we sure finishBundle is called frequently enough?

Yes, for streaming pipeline. The bundle size is sufficiently small.

> Are we sure this code will get invoked when there are failures to write to BQ?

Yes, there's a test case for that.